vermouth/tests/test_graph_utils.py

```diff
@@ -0,0 +1,783 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains unittests for the functions in vermouth.graph_utils.
+"""
+
+
+from pprint import pprint
+
+import networkx as nx
+import pytest
+import vermouth
+
+# no-member because module networkx does indeed have a member isomorphism;
+# too-many-arguments because some tests just need a lot of data.
+# pylint: disable=no-member, too-many-arguments
+
+
+def make_into_set(iter_of_dict):
+    """
+    Convenience function that turns an iterator of dicts into a set of
+    frozenset of the dict items.
+    """
+    return set(frozenset(dict_.items()) for dict_ in iter_of_dict)
+
+
+def basic_molecule(node_data, edge_data=None):
+    """
+    Construct a simple Molecule based with specified nodes and edges.
+    """
+    if edge_data is None:
+        edge_data = {}
+
+    mol = vermouth.Molecule()
+    for idx, node in enumerate(node_data):
+        mol.add_node(idx, **node)
+    for (idx, jdx), data in edge_data.items():
+        mol.add_edge(idx, jdx, **data)
+    return mol
+
+
+@pytest.mark.parametrize('node_data_in,expected_node_data', [
+    ([], []),
+    ([{'atomname': 'H3'}],
+     [{'atomname': 'H3', 'element': 'H'}]),
+    ([{'atomname': '1H3'}],
+     [{'atomname': '1H3', 'element': 'H'}]),
+    ([{'atomname': 'H3'}, {'atomname': '1H3'}],
+     [{'atomname': 'H3', 'element': 'H'}, {'atomname': '1H3', 'element': 'H'}]),
+    ([{'atomname': 'Cl1', 'element': 'Cl', 'attr': None},
+      {'atomname': '31C3'}],
+     [{'atomname': 'Cl1', 'element': 'Cl', 'attr': None},
+      {'atomname': '31C3', 'element': 'C'}]),
+    ([{'element': 'Cl'}, {'atomname': '31C3'}],
+     [{'element': 'Cl'}, {'atomname': '31C3', 'element': 'C'}]),
+])
+def test_add_element_attr(node_data_in, expected_node_data):
+    """
+    Tests for the function ``add_element_attr``.
+    """
+    mol = basic_molecule(node_data_in)
+    vermouth.graph_utils.add_element_attr(mol)
+    expected = basic_molecule(expected_node_data)
+    assert mol.nodes(data=True) == expected.nodes(data=True)
+
+
+@pytest.mark.parametrize('node_data_in,exception', [
+    ([{'atomname': '1234'}], ValueError),
+    ([{'peanuts': '1H3'}], ValueError),
```

Comment:
I know now why these case raise errors, but a quick comment at the end of the lines will save some time down the road. Something like "no atomname attribute", and "no alpha character".